### PR TITLE
[Android] Add `forceGetLocation` filed to get AssetEntity

### PR DESCRIPTION
**Pull request by lantah-1**
Use scenario: When retrieving List<Asset Entity> object, you want to return the location information of all Asset Entity objects directly, instead of iterating through the List to retrieve it individually through get Lat Lng Async method, which is time-consuming

使用场景：获取List<AssetEntity>对象时，想要直接返回所有AssetEntity对象的地理位置信息，而不是遍历list去单独的通过getLatLngAsync方法去获取，那样很耗时。

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,3 +15,5 @@ fvm
 doc/
 .env
 *.iml
+
+.fvm
```
Comment (AlexV525) [inline .gitignore:19]: Remove this.

File: android/src/main/kotlin/com/fluttercandies/photo_manager/core/utils/IDBUtils.kt [hunk 2]
```diff
@@ -94,9 +94,10 @@ interface IDBUtils {
         size: Int,
         requestType: Int = 0,
         option: FilterOption,
+        forceGetLocation: Boolean
     ): List<AssetEntity>
 
-    fun getAssetEntity(context: Context, id: String): AssetEntity?
+    fun getAssetEntity(context: Context, id: String,forceGetLocation: Boolean = false): AssetEntity?
```
Comment (AlexV525) [inline android/src/main/kotlin/com/fluttercandies/photo_manager/core/utils/IDBUtils.kt:100]: `forceGetLocation` should be inside the `FilterOption`. Also, the code is not well-formatted.

File: android/src/main/kotlin/com/fluttercandies/photo_manager/core/utils/AndroidQDBUtils.kt [hunk 3]
```diff
@@ -248,7 +254,9 @@ object AndroidQDBUtils : IDBUtils {
         val modifiedDate = cursor.getLong(MediaStore.MediaColumns.DATE_MODIFIED)
         val orientation: Int = cursor.getInt(MediaStore.MediaColumns.ORIENTATION)
         val relativePath: String = cursor.getString(MediaStore.MediaColumns.RELATIVE_PATH)
-        if ((width == 0 || height == 0)
+        var lat = 0.0
+        var log = 0.0
+        if ((width == 0 || height == 0 || forceGetLocation)
```
Comment (AlexV525) [inline android/src/main/kotlin/com/fluttercandies/photo_manager/core/utils/AndroidQDBUtils.kt:259]: The request of `ExifInterface.latLong` breaks the condition of obtaining the `width` and `height`.

**Review by AlexV525 — CHANGES_REQUESTED** [review]

**Comment by AlexV525** [comment]
The PR itself seems ineffective. EXIF should depend on permissions on recent Android versions, and the request will make the loading complexity increase from O(n) to O(n^2).

File: android/src/main/kotlin/com/fluttercandies/photo_manager/core/utils/AndroidQDBUtils.kt (hunk 3 above)
Comment (lantah-1) [reply android/src/main/kotlin/com/fluttercandies/photo_manager/core/utils/AndroidQDBUtils.kt]: Suppose there are 100,000 pictures in the album, and the pictures with geographic location information need to be found out from them. According to the previous logic, 100,000 pictures need to be recycled first, and then 100,000 times to obtain the exIF of the picture. Now, the need Geo field is used to control the extraction of exIF information when the picture is taken out, which saves the cycle time

假设相册有10万张图片，需要从其中找出有地理位置信息的图片，按照之前的逻辑就需要先循环取出10万张图片，再循环10万次去获取图片的exif,而现在通过needGeo字段去控制在取出图片的同时取出exif信息，节省了循环的时间

**Comment by lantah-1** [comment]
> The PR itself seems ineffective. EXIF should depend on permissions on recent Android versions, and the request will make the loading complexity increase from O(n) to O(n^2).

Suppose there are 100,000 pictures in the album, and the pictures with geographic location information need to be found out from them. According to the previous logic, 100,000 pictures need to be recycled first, and then 100,000 times to obtain the exIF of the picture. Now, the need Geo field is used to control the extraction of exIF information when the picture is taken out, which saves the cycle time

假设相册有10万张图片，需要从其中找出有地理位置信息的图片，按照之前的逻辑就需要先循环取出10万张图片，再循环10万次去获取图片的exif,而现在通过needGeo字段去控制在取出图片的同时取出exif信息，节省了循环的时间

**Comment by AlexV525** [comment]
> Suppose there are 100,000 pictures in the album, and the pictures with geographic location information need to be found out from them. According to the previous logic, 100,000 pictures need to be recycled first, and then 100,000 times to obtain the exIF of the picture. Now, the need Geo field is used to control the extraction of exIF information when the picture is taken out, which saves the cycle time

This only saves callers, not the total time. Also, I don't see you're handling the permission here.

File: android/src/main/kotlin/com/fluttercandies/photo_manager/core/entity/FilterOption.kt [hunk 4]
```diff
@@ -30,6 +30,7 @@ class FilterCond {
     var isShowTitle = false
     lateinit var sizeConstraint: SizeConstraint
     lateinit var durationConstraint: DurationConstraint
+    var isNeedGeo = false
```
Comment (AlexV525) [inline android/src/main/kotlin/com/fluttercandies/photo_manager/core/entity/FilterOption.kt:33]: Consist variables name everywhere.

File: android/src/main/kotlin/com/fluttercandies/photo_manager/core/utils/AndroidQDBUtils.kt [hunk 5]
```diff
@@ -248,21 +254,32 @@ object AndroidQDBUtils : IDBUtils {
...
+                if (getGeo) {
+                    it.latLong?.apply {
+                        lat = this[0]
+                        log = this[1]
                     }
+                    getGeo = false
+                }
+            }
+        }
+        if (getGeo) {
```
Comment (AlexV525) [inline android/src/main/kotlin/com/fluttercandies/photo_manager/core/utils/AndroidQDBUtils.kt:277]: The EXIF interface will be created again here. The situation is even worse.

File: lib/src/filter/filter_options.dart [hunk 6]
```diff
@@ -13,6 +13,7 @@ import '../internal/enums.dart';
 class FilterOption {
   const FilterOption({
     this.needTitle = false,
+    this.needGeo = false,
```
Comment (AlexV525) [inline lib/src/filter/filter_options.dart:16]: Better be `needLatLng`.

File: lib/src/internal/plugin.dart [hunk 7]
```diff
@@ -123,6 +123,7 @@ class PhotoManagerPlugin with BasePlugin, IosPlugin, AndroidPlugin {
     required int start,
     required int end,
     required FilterOptionGroup optionGroup,
+    bool forceGetLocation = false,
```
Comment (AlexV525) [inline lib/src/internal/plugin.dart:126]: Remove useless code.

**Review by AlexV525 — CHANGES_REQUESTED** [review]

**Comment by AlexV525** [comment]
The solution is not in a good state IMO. Please file a new PR after the design is more solid and clear. Thanks.

**Pull request closed without merging**